packages/react-server/core/components/RootElement.js

```diff
@@ -103,7 +103,7 @@ RootElement.getRootElementAttributes = function(element) {
 		'id',
 		'style',
 	].forEach(k => props[k] && (attrs[k] = props[k]));
-
+	attrs.tagName = props.tagName || 'div';
```

Comment:
I wouldn't bundle this in with `attrs`.  That's really meant to just be HTML attributes.

The tag name is a different beast.